.github/workflows/pull-soljson.yml

```diff
@@ -0,0 +1,120 @@
+          # For some reason git stages all files for deletion when you use --no-checkout
+          git reset HEAD --quiet
+
+      - name: Check out only the files we're going to actually use
+        run: |
+          git checkout -- update package.json
+          git checkout -- bin/list.json
+          git checkout -- bin/list.js
+          git checkout -- bin/list.txt
+          git checkout -- wasm/list.json
+          git checkout -- wasm/list.js
+          git checkout -- wasm/list.txt
+
+          # FIXME: The update script crashes if there isn't at least one release already stored.
+          git checkout -- update bin/soljson-latest.js
+          git checkout -- update wasm/soljson-latest.js
+          git checkout -- update bin/soljson-v0.3.6+commit.3fc68da5.js
+          git checkout -- update wasm/soljson-v0.3.6+commit.3fc68da5.js
+
+      - name: Get job info using CircleCI API
+        run: |
+          job_info_endpoint="https://circleci.com/api/v1.1/project/github/ethereum/solidity?filter=successful&limit=100"
+          job_info=$(
+            curl --silent --show-error "$job_info_endpoint" |
+            jq '[ .[] | select (.workflows.job_name == "'${JOB_NAME}'") ] | sort_by(.build_num) | last'
+          )
+
+          build_num=$(echo "$job_info" | jq ".build_num")
+
+          commit_hash=$(echo "$job_info" | jq --raw-output ".vcs_revision" | head --bytes 8)
+          # FIXME: CircleCI API provides a tag name only if it's directly on the current commit.
+          solidity_version="[version-placeholder]"
+          nightly_version="${solidity_version}+commit.${commit_hash}"
+          version_already_exists="$(test -f "bin/soljson-$nightly_version.js" && echo true || true)"
+
+          echo "::set-env name=NIGHTLY_VERSION::${nightly_version}"
+          echo "::set-env name=VERSION_ALREADY_EXISTS::${version_already_exists}"
+          echo "::set-env name=BUILD_NUM::${build_num}"
+
+      - name: Get artifact URL using CircleCI API
+        if: "!env.VERSION_ALREADY_EXISTS"
+        run: |
+          artifacts_endpoint="https://circleci.com/api/v1.1/project/github/ethereum/solidity/${BUILD_NUM}/artifacts"
+          artifact_url=$(
+            curl --silent --show-error "$artifacts_endpoint" |
+            jq --raw-output '.[].url'
+          )
+
+          echo "::set-env name=ARTIFACT_URL::${artifact_url}"
+
+      - name: Download the binary from CircleCI
+        if: "!env.VERSION_ALREADY_EXISTS"
+        run: |
+          # TMP:
+          #curl --silent --show-error --location --create-dirs --output "$NIGHTLY_PATH" "$ARTIFACT_URL"
+          curl --silent --show-error --location --remote-name "$ARTIFACT_URL"
+          test -f soljson.js
+
+          cp soljson.js "bin/soljson-$NIGHTLY_VERSION.js"
+          mv soljson.js bin/soljson-nightly.js
+
+          # TODO: Add a symlink under wasm/ if it's a tagged commit
+          # [ "$TRAVIS_BRANCH" = release ] && ln -sf ../bin/"$NEWFILE" ./wasm/"$NEWFILE"
```

Comment:
Just for the sake of my argument against fetching this from CircleCI - almost all this can mostly be replaced by something like this, if we rebuild:
```
      - uses: actions/checkout@v2
        with:
          repository: 'ethereum/solidity'
          ref: 'develop'
          path: 'solidity'
      - name: Build soljson.js
        run: |
          cd solidity
          scripts/build_emscripten.sh
          nightly_version=v$(cat CMakeLists.txt | grep 'set(PROJECT_VERSION' | sed -e 's/.*set(PROJECT_VERSION \"\(.*\)\".*/\1/')-nightly.$(date --date="$(git log -1 --date=iso --format=%ad HEAD)" --utc +%Y.%-m.%-d)+commit.$(git rev-parse --short=8 HEAD)
          echo "::set-env name=NIGHTLY_VERSION::${nightly_version}"
          cp upload/soljson.js ../soljson-${nightly_version}.js
          cd ..
          rm -rf solidity
          if git ls-files --other --directory --exclude-standard | sed q1; then echo "::set-env name=VERSION_ALREADY_EXISTS::1"; fi
```
The build could of course happen to fail, if develop happens to be broken, but I don't think that's a huge problem... and it will take some time, but again I don't find that too bad actually. But yeah, just something to think about :-).

Reply:
I reused most of your suggestion in #30.